File: backend/RunHub.Domain/Entities/Distance.cs
using RunHub.Domain.Enums;

namespace RunHub.Domain.Entity
{
    public class Distance
    {
        public int DistanceId { get; set; }
        public string Name { get; set; }
        public double LengthInKilometers { get; set; }
        public string Description { get; set; }
        public int AvailableSlots { get; set; }
        public int TotalSlots { get; set; }
        public decimal Price { get; set; }
        public DistanceStatus Status { get; set; } = DistanceStatus.Active;
        public bool IsReadyToShow { get; set; } = false;

        public int RaceId { get; set; }
        public Race Race { get; set; }
        public ICollection<DistanceAttendee> DistanceAttendees { get; set; } = new List<DistanceAttendee>();
        public ICollection<Result> Results { get; set; } = new List<Result>();
    }
}

File: frontend/runHubFront/src/app/models/distance.ts
import { Profile } from "./profile";

export interface IDistanceDto {
    distanceId?: number; //?
    name: string;
    lengthInKilometers: number;
    description: string;
    availableSlots: number;
    totalSlots: number;
    price: number;
    hostUsername?: string;
    distanceAttendees?: Profile[]
    //extra property
    isGoing:boolean;
    isReadyToShow?:boolean;
    registeredUser?:number;
    status?:DistanceStatus;
    date?:Date ;
  }

  export class DistanceDto implements IDistanceDto {
    distanceId?: number;
    name: string;
    lengthInKilometers: number;
    description: string;
    availableSlots: number;
    totalSlots: number;
    price: number;
    hostUsername?: string;
    distanceAttendees: Profile[] = [];
    isGoing: boolean = false
    isReadyToShow?:boolean;
    registeredUser?:number;
    status?: DistanceStatus;
    date?: Date;



    constructor(init: IDistanceDto) {
        this.distanceId = init.distanceId ;
        this.name = init.name;
        this.lengthInKilometers = init.lengthInKilometers;
        this.description = init.description;
        this.availableSlots = init.availableSlots;
        this.totalSlots = init.totalSlots;
        this.price = init.price;
        this.hostUsername = init.hostUsername
        // this.isGoing = init.isGoing;
        this.isReadyToShow = init.isReadyToShow;
        // this.distanceAttendees = init.distanceAttendees ?? [];
        this.registeredUser = init.registeredUser;
        this.status = init.status;
        this.date = init.date
    }
}


export enum DistanceStatus {
  Active, 
  Cancelled 
}

export const distanceStatusOptions = [
  { text: 'Aktywny', value: DistanceStatus.Active },
  { text: 'Odwołane', value: DistanceStatus.Cancelled },
];
